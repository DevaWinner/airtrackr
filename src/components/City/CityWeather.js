import PropTypes from 'prop-types';

function CityWeather({ city, aqi }) {
  let weatherStatus = '';

  switch (aqi) {
    case 1:
      weatherStatus = 'Excellent';
      break;
    case 2:
      weatherStatus = 'Great';
      break;
    case 3:
      weatherStatus = 'Good';
      break;
    case 4:
      weatherStatus = 'Fair';
      break;
    case 5:
      weatherStatus = 'Poor';
      break;
    default:
      weatherStatus = 'Unknown';
  }

  const statusClassName = weatherStatus.toLowerCase().replace(' ', '-');

  return (
    <div className="city-weather flex">
      <h2>{city}</h2>
      <div className="air-quality">
        <span className="quality">Quality of air:</span>
        <span className={statusClassName}>{weatherStatus}</span>
      </div>
      <div className="quality-index">
        <span className="quality-index-header">Quality index:</span>
        <span className={statusClassName}>{aqi}</span>
      </div>
    </div>
  );
}

CityWeather.propTypes = {
  city: PropTypes.string.isRequired,
  aqi: PropTypes.number.isRequired,
};

export default CityWeather;
