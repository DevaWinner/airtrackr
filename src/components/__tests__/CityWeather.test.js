import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import CityWeather from '../City/CityWeather';

describe('CityWeather', () => {
  it('renders the city name and AQI status', () => {
    const { getByText } = render(<CityWeather city="New York" aqi={3} />);
    expect(getByText('New York')).toBeInTheDocument();
    expect(getByText('Good')).toBeInTheDocument();
  });

  it('renders the AQI value', () => {
    const { getByText } = render(<CityWeather city="Los Angeles" aqi={5} />);
    expect(getByText('Air Quality Index(AQI):')).toBeInTheDocument();
    expect(getByText('5')).toBeInTheDocument();
  });

  it('renders the correct status class name', () => {
    const { getByText } = render(<CityWeather city="Chicago" aqi={2} />);
    expect(getByText('Great')).toHaveClass('great');
  });

  it('should render Cities component', () => {
    const city = 'Cairo';
    const aqi = 1;
    render(<CityWeather city={city} aqi={aqi} />);
    expect(CityWeather).toMatchSnapshot();
  });

  it('should render Cities component', () => {
    const city = 'Cairo';
    const aqi = 1;
    render(<CityWeather city={city} aqi={aqi} />);
    expect(CityWeather).toMatchSnapshot();
  });
});
