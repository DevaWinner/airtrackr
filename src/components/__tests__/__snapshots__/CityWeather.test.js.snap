// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeather should render Cities component 1`] = `[Function]`;

exports[`CityWeather should render CityWeather component 1`] = `[Function]`;
